src/30_Advanced/SeatMap.html

```diff
@@ -0,0 +1,185 @@
+        bottom: 60px;
+        right: 0px;
+        z-index: 1;
+        border-color: Transparent;
+        background-color: Transparent;
+        margin: 8px;
+        width: 40px;
+        height: 40px;
+        padding: 0;
+        border-width: 0;
+        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.25);
+        background-color: white;
+        border-radius: 3px;
+      }
+      .opaque {
+        opacity: 0.5;
+        filter: alpha(opacity=50);
+      }
+      .seat-map-svg {
+        box-sizing: border-box;
+        padding-top: 30px;
+      }
+      #pan-zoom {
+        display: initial;
+      }
+      #pan-zoom > svg {
+        left: 0;
+        right: 0;
+        margin: auto;
+      }
+      .list-container {
+        position: relative;
+        height: 70vh;
+      }
+      .ampstart-card {
+        min-width: 100%;
+        padding: 1rem;
+      }
+    </style>
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style>
+    <noscript>
+      <style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style>
+    </noscript>
+
+  </head>
+  <body>
+
+    <!-- Here we are using an `amp-state` called `seatmap`.
+      `scale` variable is used to keep track of the scale of the content being zoomed using by `amp-pan-zoom`.
+      `zoom` variable is used to keep track of the seat map is in a zoomed state or not. This variable is used to
+    decide whether to show a + or - button.
+    `selectedSeats` variables keeps a  list of selected seat ids.  -->
+    <amp-state id="seatmap">
+      <script type="application/json">
+        {
+          "scale": 1,
+          "selectedSeats": [],
+          "zoom": 0
+        }
+      </script>
+    </amp-state>
+
+    <h1>Seat map</h1>
+    <!-- You can implement a seat map by using `amp-list`, `amp-bind` and `amp-pan-zoom`.
+    `amp-list` is used to keep the availability of the seatmap up to date.
+    `amp-bind` is used to update the status of the seats: when a seat is selected or unselected we update the color and 
+    we add the seat id to the `seatmap.selectedSeats` variable. 
+    `amp-pan-zoom` is used to make possible to pan & zoom on mobile devices.
+    We are using the `transformEnd` action from [amp-pan-zoom](https://www.ampproject.org/docs/reference/components/amp-pan-zoom)
+    to update the value of transform.scale variable.
+    -->
+     <div class="list-container">
```

Comment:
this is the way I got it working

Reply:
in a real case you wouldn't know upfront the height of the seatmap which comes from the endpoint, this is why I don't want to use fixed-height.
Unfortunately responsive will not work either.

Reply:
But doesn't amp-pan-zoom solve this? The amp-list defines the layout (e.g. responsive) and then amp-pan-zoom fills the amp-list? 

If this doesn't work, please explain it in the doc.

Reply:
the problem with responsive is similar with fixed-height: you need to indicate a width and height that you may not know.

Reply:
I think in the amp-pan-zoom sample we can show all the possible layouts. When the dimensions are not known, I think this is the best solution.

Reply:
But you do specify a specific height via CSS: 

```
         height: 50vh;
```

Reply:
yes so that it will be inherited. This will be up to a developer to decide how to do it in the css, an alternative is to calculate the available space left and give the seatmap all the space. In this case as we don't have some many other things in the page just using half of it would work

Reply:
Can you add a comment explaining why you're using a fill layout then.

Reply:
done!